llvm/lib/Transforms/Yk/ControlPoint.cpp

```diff
@@ -48,8 +48,8 @@
 //
 // YKFIXME: The tracing logic is currently over-simplified:
 //
-//  - A JIT location is assumed to be a simple integer program counter. It
-//    should be a `ykrt::Location`.
+//  - The address of `YkLocation` instances are used for identity, but they are
```

Comment:
> The address of `YkLocation` instances are used for identity

This isn't true, and indeed is immediately contradicted by "freely moved by the user" :)

Users are free to move `YkLocation`s as often as they want. What they *must* do is always pass the same `YkLocation` *value* for the same program location: the address that's used as an identifier is stored *in* the (opaque) `YkLocation`.

Reply:
Yes. This list is a list of deficiencies that should be fixed :P

Reply:
Ah, can we clarify that these are flaws, not just over simplifications?

Reply:
Sure. Try 25f4a59.